### ISSUE TITLE
`isinstance(_, Callable)` reports `[arg-type]`

### ISSUE THREAD

**Issue by jorenham** [python/mypy#18796]
**Bug Report**

`isinstance(_, collections.abc.Callable)` is allowed at runtime, but is reported

(A clear and concise description of what the bug is.)

**To Reproduce**

```python
from collections.abc import Callable
assert isinstance(int, Callable)
```

**Expected Behavior**

for `isinstance(_, collections.abc.Callable)` to behave identically to `callable(_)`:

- no error
- `_: T` is narrowed to `_: <callable subtype of T>`

**Actual Behavior**

```
main.py:2: error: Argument 2 to "isinstance" has incompatible type "<typing special form>"; expected "_ClassInfo"  [arg-type]
```

https://mypy-play.net/?mypy=latest&python=3.13&gist=463e1edfbfc8cefc268fce8544b0b49b

**Comment by A5rocks** [comment]
As a work around the `callable` builtin narrows. (You're probably aware though)

**Comment by jorenham** [comment]
> As a work around the `callable` builtin narrows. (You're probably aware though)

I'm indeed aware, but others might not, so it's good that you mention it 👍🏻
